src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -175,81 +175,81 @@ def _find_resources(self, node, find_slots, ranks_per_slot, cores_per_slot,
                thread count and using physical core IDs for process placement?
         '''
 
-      # self._log.debug('find on %s: %s * [%s, %s]', node['uid'], )
+        slots = list()
 
-        # check if the node can host the request
-        free_cores = node['cores'].count(rpc.FREE)
-        free_gpus  = node['gpus'].count(rpc.FREE)
-        free_lfs   = node['lfs']
-        free_mem   = node['mem']
+        # find at most `n_slots`
+        while len(slots) < n_slots:
```

Comment:
seems like if there are more than 1 slot, then we go over the same resources (cores and gpus), originally we kept track of the current index for cores/gpus while were going through `alc_slots`

example from the test run (2 ranks with 31 cores and 2 gpus per rank)

<details>

```
> cat task.000000/task.000000.sl
[
    {
        "cores": [
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            22,
            23,
            24,
            25,
            26,
            27,
            28,
            29,
            30,
            31,
            32
        ],
        "gpus": [
            [
                0,
                1.0
            ],
            [
                1,
                1.0
            ]
        ],
        "lfs": 0,
        "mem": 0,
        "node_index": 0,
        "node_name": "x3005c0s31b1n0"
    },
    {
        "cores": [
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            22,
            23,
            24,
            25,
            26,
            27,
            28,
            29,
            30,
            31,
            32
        ],
        "gpus": [
            [
                0,
                1.0
            ],
            [
                1,
                1.0
            ]
        ],
        "lfs": 0,
        "mem": 0,
        "node_index": 0,
        "node_name": "x3005c0s31b1n0"
    }
]
```

</details>

p.s., should we have `_RO` for cores and gpus?

Reply:
Thanks, this is fixed now!

Reply:
it allocates correctly now, but something wrong with tags
```python
for idx in range(NUM_NODES):
    tds.append(
        rp.TaskDescription({
            ...
            'tags': {'colocate': str(idx),
                     'exclusive': True},
```
with this code, each of this tasks should be placed on their own node,
but currently it was placed on the same (first) node